### ISSUE TITLE
[HOLD for payment 2024-08-05] [$1000] Standardize on single fontFamily names across platforms

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#13772]
# Problem
Coming from [this discussion](https://github.com/Expensify/App/pull/13714#discussion_r1054873229), [react-native-asset](https://github.com/unimonkiez/react-native-asset) let's us easily link and manage assets, including fonts. However, it fails to provide a cross platform way to use font typeface modifiers such as `fontWeight` and `fontStyle` in combination with a custom `fontFamily` name, in both iOS and Android, which in turn results in keeping multiple font-families such as, `ExpensifyNewKansas-Medium`, `ExpensifyNewKansas-MediumItalic` instead of a single `ExpensifyNewKansas` font-family.

iOS seems to work out of the box as long as we use the postScript name for the font. However, android uses the font filename which results in the different `fontFamilies` above.

# Why is this important
Ease of importing new fonts and keeping/maintinaing a single font type-face.

# Solution
Update [react-native-asset](https://github.com/unimonkiez/react-native-asset) to support [XML fonts](https://github.com/jsamr/react-native-font-demo#android) in android and auto-group font families into XML fontFamilies.

cc @roryabraham 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0165fabbe62a87ed20</li>
        <li>Upwork Job ID: 1641731454588358656</li>
        <li>Last Price Increase: 2023-03-31</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sakluger / @zanyrenney</details>

**Comment by roryabraham** [comment]
cc @Luke9389 for visibility

**Comment by grgia** [comment]
nice!

**Comment by Luke9389** [comment]
Love it! Thanks y'all

**Comment by roryabraham** [comment]
No update yet

**Comment by roryabraham** [comment]
Sorry for the delay here. I'm going to make this a weekly, but my plan here is to:

1. Create a couple simple demo React Native projects to show the difference between the "old" way of bundling fonts that we use today vs the "new" way (i.e: Android xml fonts)
1. Then create an issue in https://github.com/unimonkiez/react-native-asset to show the "before" and "after" that explains what we would expect react-native-asset to do, given font assets in the top-level assets directory.
1. Once that is done, we can re-group, but it's likely I would want to pass this off to an expert contractor (or even better, hire the react-native-asset maintainer directly) to do the actual implementation. I _think_ I could implement it, but I just don't know when I'll have enough time.

**Comment by Luke9389** [comment]
I think that sounds great @roryabraham. Thanks for the update

**Comment by roryabraham** [comment]
No update this week :(

**Comment by roryabraham** [comment]
No update again

**Comment by roryabraham** [comment]
Sorry, still no update here.

**Comment by roryabraham** [comment]
No update, won't prioritize until after ECX at the earliest. If we could very clearly state our goal here we could pass this off to an agency, which has been my intention since the start

**Comment by roryabraham** [comment]
I've held onto this for too long and I've been struggling to make time for this. Given that I have several other higher-priority initiatives, I'm going to drop this for now.

**Comment by luacmartins** [comment]
Should we make this external? Other contributors can propose a fix to [react-native-asset](https://github.com/unimonkiez/react-native-asset)

**Comment by luacmartins** [comment]
Asked callstack [here](https://expensify.slack.com/archives/C03UK30EA1Z/p1679351442650589)

**Comment by fabioh8010** [comment]
Hi, I'm Fábio from Callstack and I would like to work on this issue.

**Comment by fabioh8010** [comment]
**Update:** Regarding to this issue, I will start investigation today and will keep you updated here once I have a proposal.

**Comment by luacmartins** [comment]
Thanks for the update @fabioh8010!

**Comment by fabioh8010** [comment]
**Update:** Did some investigation and tests to validate the idea, I expect to provide a proposal on Monday.

**Comment by fabioh8010** [comment]
Update: Proposal being reviewed internally, also exploring one alternative solution.

**Comment by fabioh8010** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

iOS and Android have different ways of handling custom fonts on RN. While iOS has a more robust way to "query" the fonts available to use, in Android we have to rely on the font's file name to be able to use that custom font. In this way, it's difficult to have a cross-platform configuration that satisfies both platforms.

### What is the root cause of that problem?

As stated in this issue, Android has a different logic to manage fonts related to iOS, requiring the developer to rely on the font file's name to be able to use the font properly in the application.

In Android, we have the [ReactFontManager.java](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java) file that manages font typeface's cache and registry, and there is a logic to find the correct font file based on the `fontFamily` provided, an extension and the file extension itselt (.ttf or .otf). In fact, [there is indeed a logic](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java#L39) to determine if the font is bold or italic based on its name. For example, if you set `fontFamily` to `Lato` and `fontWeight` to `bold` and you have a font file called `Lato_bold.ttf`, the logic will correctly choose this file to use for the bold font you specified.

The problem is that this logic is flawed as it only works if the font file name have a specific and uncommon format, and it don't cover other font weights.

### What changes do you think we should make in order to solve the problem?

We propose to go with the CLI improvement, making it execute the steps described [here](https://github.com/jsamr/react-native-font-demo) in an automatic way. We are going to do this in the following steps:
1. When the CLI is run, search for the font files that were placed in the project.
2. For each font file:
   1. Run `fontname` npm package or similar tool to get the font’s information.
   2. Get the font's information like font style and weight, by either:
      1. Reading the file metadata and extracting this info (if possible).
      2. Guessing the font style and weight by the font's file name.
3. When all fonts are scanned, build a map with all different font families found.
4. For each font family:
   1. Rename the corresponding font files and copy to android’s folder.
   2. Create a XML file with all variants for this font family (we can use `fast-xml-parser` npm package).
   3. Use regular expressions in the `MainApplication.java` file to find and insert the import and the `addCustomFont()` method to register that font family.
5. Make a PR to react-native-asset library.
6. When merged, use this tool in Expensify's project and make a PR with the changes.

### What alternative solutions did you explore? (Optional)

We thought about making an improvement directly in the RN repo, improving `ReactFontManager`'s logic to detect the fonts in a better way than the current one (using common suffixes for these fonts), but we ended up sticking with the CLI solution as we think it will provide a safer approach to use the fonts in a cross-platform way.

**Comment by roryabraham** [comment]
@fabioh8010 That sounds great to me 👍🏼

**Comment by roryabraham** [comment]
I've contributed to react-native-asset before so can help review.

**Comment by fabioh8010** [comment]
Great! I'll start development tomorrow and will keep you updated here.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~0165fabbe62a87ed20

**Comment by MelvinBot** [comment]
Triggered auto assignment to @sakluger (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mollfpr (`External`)

**Comment by MelvinBot** [comment]
Current assignees @roryabraham and @luacmartins are eligible for the External assigner, not assigning anyone new.

**Comment by luckchain007** [comment]
In order to achieve cross-platform compatibility for custom fontFamily names with fontWeight and fontStyle modifiers, we can follow the steps below:

Rename the font files to match the desired fontFamily name, for example, "ExpensifyNewKansas-Regular" for the regular weight.

In the react-native.config.js file, add the following code snippet:

module.exports = {
assets: ['./src/assets/fonts/'],
ios: {},
android: {
sourceDir: './android',
fontFamily: 'ExpensifyNewKansas',
fontWeight: {
Regular: '400',
Medium: '500',
Bold: '700',
},
fontStyle: {
Italic: 'italic',
},
},
};

Here, we have added "fontFamily", "fontWeight", and "fontStyle" properties to the Android configuration.

Modify the styles as follows:
const styles = StyleSheet.create({
textRegular: {
fontFamily: Platform.OS === 'android' ? 'ExpensifyNewKansas-Regular' : 'ExpensifyNewKansas',
fontWeight: Platform.OS === 'android' ? '400' : 'normal',
fontStyle: 'normal',
},
textMedium: {
fontFamily: Platform.OS === 'android' ? 'ExpensifyNewKansas-Medium' : 'ExpensifyNewKansas',
fontWeight: Platform.OS === 'android' ? '500' : 'bold',
fontStyle: 'normal',
},
textBoldItalic: {
fontFamily: Platform.OS === 'android' ? 'ExpensifyNewKansas-Bold' : 'ExpensifyNewKansas-BoldItalic',
fontWeight: Platform.OS === 'android' ? '700' : 'bold',
fontStyle: 'italic',
},
});

In the above code, we use the Platform module to differentiate between iOS and Android and set the appropriate font family, fontWeight, and fontStyle properties. This allows us to use a single fontFamily for all weight and style variations, including custom names with modifiers.

By following these steps, we can achieve cross-platform compatibility with custom fontFamily names in combination with fontWeight and fontStyle modifiers.
I wish hope you are doing well
luckhole19971119@gmail.com

**Comment by MelvinBot** [comment]
📣 @luckchain007! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by luckchain007** [comment]
Or, While react-native-asset provides a convenient way to manage assets, including fonts, it does not offer a cross-platform solution for using font typeface modifiers with custom fontFamily names. This can result in having to create multiple font families, each with a different name, in order to use different font weights and styles.

In iOS, using the postScript name for the font works out of the box and allows you to use font typeface modifiers with a custom fontFamily name. However, in Android, the font filename is used instead of the postScript name, which can lead to the creation of multiple font families.

To achieve a consistent font family name across both iOS and Android, you can modify the font files to include the necessary information in their filenames. For example, you can add "-Regular", "-Medium", "-Bold", or "-Italic" to the font filename to indicate the font weight or style. This will ensure that the correct font is used on both platforms, without the need to create multiple font families.

Alternatively, you can use a third-party library, such as react-native-vector-icons, which offers cross-platform support for font typeface modifiers with custom fontFamily names. This library provides a wide range of icons, as well as support for custom fonts, and allows you to easily use font typeface modifiers with a custom fontFamily name on both iOS and Android.

**Comment by luacmartins** [comment]
@luckchain007 sorry, this issue is not open to proposals. I just added the External label to help me manage the issue. @fabioh8010 is already working on a solution.

**Comment by luckchain007** [comment]
> @luckchain007 sorry, this issue is not open to proposals. I just added the External label to help me manage the issue. @fabioh8010 is already working on a solution.

Good

**Comment by fabioh8010** [comment]
**Update:** Ended up starting development today because I was working in other issues.

**Comment by luckchain007** [comment]
Thanks!

**Comment by fabioh8010** [comment]
**Update:** Currently working on step 2, used `fontname` to retrieve font's information and looking a way to extract font style and weight.

**Comment by sakluger** [comment]
@fabioh8010 just to confirm, there's no draft PR yet, right? Just wanted to make sure I didn't miss anything as I monitor this issue.

**Comment by luckchain007** [comment]
Yes, that is correct.

**Comment by fabioh8010** [comment]
@sakluger Yes, I'm still working on this feature. Currently trying to find a better way to extract font style and weight rather than relying on font family info or file name, which could have more risks.

**Comment by luckchain007** [comment]
Cool, thanks!

**Comment by luacmartins** [comment]
@fabioh8010 any updates here?

**Comment by MelvinBot** [comment]
@sakluger, @fabioh8010, @luacmartins, @mollfpr, @roryabraham Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by fabioh8010** [comment]
**Update:** Currently working on step 4, XML parsing. Will be able to raise a PR soon.

**Comment by roryabraham** [comment]
Let's continue updates in https://github.com/unimonkiez/react-native-asset/issues/48. @fabioh8010 if you could post a summary of our problem and proposed solution in that issue, that would be great to get early feedback from the maintainer, who has already commented on the issue.

Thanks!

**Comment by luacmartins** [comment]
Latest update [here](https://github.com/unimonkiez/react-native-asset/issues/48#issuecomment-1515374653)

**Comment by luckchain007** [comment]
May I interrupt here, @luacmartins

**Comment by MelvinBot** [comment]
📣 @luckchain007! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/231263719-a8a3b7d3-36a0-4d7f-8ad4-cebe1f560fa4.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by luacmartins** [comment]
>May I interrupt here, @luacmartins

@luckchain007 I'm confused by this message

**Comment by roryabraham** [comment]
@fabioh8010 is OOO until May 4th I think

**Comment by fabioh8010** [comment]
**Update:** Waiting for maintainer's confirmation about our approach in order to finish the feature.

**Comment by sakluger** [comment]
@fabioh8010 any confirmation or updates yet?

**Comment by fabioh8010** [comment]
@sakluger I asked him to elaborate more about his idea in order to finish the feature, and also asked him to review my PR meanwhile (to speed up things) because the last missing part is more like a complementary thing, the core feature is already developed and waiting for his review.

**Comment by fabioh8010** [comment]
**Update:** Got the review and the answer to my questions for now, will address the requested changes.

**Comment by fabioh8010** [comment]
**Update:** Addressed the changes and final implementation requested by maintainer. Tested with Expensify fonts and it's working well.

![Screenshot 2023-05-20 at 17 56 51](https://github.com/Expensify/App/assets/20051562/c97228af-91c6-4a4c-aa6f-3dd2a7230b23)

**Comment by luacmartins** [comment]
Nice! @fabioh8010 what are the next steps?

**Comment by fabioh8010** [comment]
@luacmartins It would be:

1. Wait for the PR to be merged and a new release be made.
2. Use the tool and refactor our project to standardize font family usage.

If the issue has more urgency I think we can skip the first step and use the tool (based on my PR) right away.

**Comment by luacmartins** [comment]
I think we're ok waiting for it since this is more of a clean up. Thanks for the update!

**Comment by fabioh8010** [comment]
**Update:** Asked the reviewer if he had a chance to take a look at the PR.

**Comment by luacmartins** [comment]
Still waiting for the external PR to be merged.

**Comment by fabioh8010** [comment]
Still waiting for maintainer's response.

**Comment by roryabraham** [comment]
Still waiting for maintainer's response

**Comment by fabioh8010** [comment]
**Update:** Asked maintainer for feedback again.

**Comment by luacmartins** [comment]
Still waiting on maintainers

**Comment by luacmartins** [comment]
Still waiting for a maintainer review here - https://github.com/unimonkiez/react-native-asset/pull/52

**Comment by luacmartins** [comment]
Still no updates from the maintainer

**Comment by roryabraham** [comment]
Bumped the maintainer. There hasn't been any activity since May, so I'm wondering if we should open an issue to ask if they're interested in passing the project off or adding additional maintainers

**Comment by luacmartins** [comment]
Still waiting on maintainer - https://github.com/unimonkiez/react-native-asset/pull/52

**Comment by roryabraham** [comment]
At this point, should we just use a patch to reap the benefits of @fabioh8010's work in the short-term? In the long term, maybe we should open an issue to ask if they want to hand that repo off to the Expensify or Callstack org?

**Comment by luacmartins** [comment]
I don't feel passionate either way. I think this is a low priority so we could wait, but I do see the argument to just get this done already. Up to you @fabioh8010

**Comment by fabioh8010** [comment]
@roryabraham @luacmartins I would suggest to fork a repo in Expensify org and include my changes, or use my PR as dependency in `package.json`. I think patching the whole PR will create a gigantic file unnecessarily.

From CK's side, I can share with you that I have been looking for ways to try to fix this natively and if I manage to find a solution, we (CK) will propose a RFC to the community since it's a very old problem and a pain between developers.

**Comment by fabioh8010** [comment]
BTW I requested another review to the maintainer instead of commenting, let's see if this makes some difference.

**Comment by luacmartins** [comment]
Should we go with a patch?

**Comment by fabioh8010** [comment]
Actually I think patching is not even doable because my PR adds new dependencies to the library, and `patch-package` is ran after installing dependencies.

Since this library

1. Does not appear to be maintained anymore
2. The changes itself are very large for a patch file (+4,087 additions and −3,290 deletions)
3. Includes changes in `package.json` (added libraries), which I strongly believe is not designed to be handled by `patch-package`, or would at least require some changes in installation flow
4. It's a complete feature and due to point 1, it's likely going to stay for a long time in our repo if patched.

Could we consider using creating a fork for this particular case? Patch files are intended to fix issues or small features temporally, and with an increasing list of patch files I believe we should start avoid adding more ones whenever possible.

<img width="494" alt="Screenshot 2023-09-04 at 12 18 25" src="https://github.com/Expensify/App/assets/20051562/944080e2-f6cd-468e-a7cc-94b5d28aab5f">

**Comment by roryabraham** [comment]
I would love to get in touch with the maintainer and ask if they want to transfer ownership of the repo or at least add some more maintainers rather than forking. We know from experience that using forks comes with a lot of overhead

**Comment by fabioh8010** [comment]
@roryabraham Sure, I understand! If Expensify doesn't have immediate plans to take this on, I'd be happy to step up as a maintainer or owner of the package, since it would be helpful for our side to be maintaining this library and making future plans for it. WDYT?

**Comment by roryabraham** [comment]
Let's create an issue in the repo 🙇🏼

**Comment by sakluger** [comment]
Hey @roryabraham @fabioh8010 what's the next step here?

**Comment by fabioh8010** [comment]
@sakluger @roryabraham Sorry for the late, I'm going to create an issue in the library's repo today 🙂

**Comment by fabioh8010** [comment]
@roryabraham I've created the issue -> https://github.com/unimonkiez/react-native-asset/issues/57

**Comment by fabioh8010** [comment]
FYI I will be on vacations starting tomorrow for the next two weeks and will return on Oct 23!

**Comment by luacmartins** [comment]
@fabioh8010 is still ooo

**Comment by fabioh8010** [comment]
Update: I'm back, there is no response from maintainer yet.

**Comment by sakluger** [comment]
@fabioh8010 any new updates?

**Comment by roryabraham** [comment]
We've waited long enough here. Let's fork react-native-asset in our org or in Callstack org

**Comment by fabioh8010** [comment]
I agree @roryabraham, I'm OOO at the moment but I will take care of this, creating a fork in CK org or maintaining it by myself. I will post an update here as soon I have any info or decision. 🙂

**Comment by fabioh8010** [comment]
Update: I will create a fork in CK org, do some more testing before releasing a new version, and when it is released, will announce it in the issue, PR and here.

cc @roryabraham

**Comment by fabioh8010** [comment]
Let's hold this one for 2 days as we are discussing an alternative option internally.

**Comment by fabioh8010** [comment]
Hey @roryabraham, I've published the [@callstack/react-native-asset](https://www.npmjs.com/package/@callstack/react-native-asset) package on NPM registry, we can now proceed with the font standartization in the project! Now, as a second phase, we (Callstack) would like to propose you to integrate this library's code directly into the [React Native CLI](https://github.com/react-native-community/cli). 

We believe that the assets linking process should be part of our CLI instead of having a separate package for that, and by adding this support in the CLI we can have the following benefits:
* No need for forked dependency in the codebase
* We have maintainers dedicated to the CLI, so we can take responsibility of the future maintenance of this feature (since it seems it is not maintained by the author anymore, even if he is still active on GitHub)
* We have a bigger community around the CLI, so it's easier to gain feedback and improve this solution for its potential needs (260 stars vs 2.2k stars on GH)
* Shipping it by default would benefit all CLI users, since there wouldn't be any additional library for assets needed

We estimate this effort in 24-40 hours and we believe this is a valuable investment in the React Native ecosystem and would greatly appreciate Expensify's support in funding this initiative. 

Here is the overview of the necessary implementation:
* Incorporation of the implementation into the CLI with the necessary dependencies
* Migration of implementation to TypeScript
* Integration into the CLI commands
* Implementation of tests (if applicable)

What do you think about this approach? Please let us know if you have any questions, thanks!

**Comment by thymikee** [comment]
Hey @roryabraham 👋 @fabioh8010 and @TMisiukiewicz reached out to me already about this and we’re willing to take this command into the CLI

**Comment by roryabraham** [comment]
Ok, that's good enough for me! Let's get started on it

**Comment by Titozzz** [comment]
That will be a great addition 🥰
Thanks

**Comment by fabioh8010** [comment]
**Update:** I've started implementation today, will be working 4h per day on this, expected ETA is Dec 22nd.

**Comment by sakluger** [comment]
Commenting to make Melvin happy. @fabioh8010 are you still on track for Dec 22?

**Comment by fabioh8010** [comment]
Hi @sakluger Unfortunately no, I ended up having to do other things with higher priority before my vacations next week. Current status is:

- Setup for the new command in the CLI done
- Integration of the necessary libraries done
- Started refactoring and migrating to TS the logic from the original library to integrate into the CLI

**Comment by fabioh8010** [comment]
Just FYI that I'm going to be OOO from 25/12 to 05/01 due to vacations. However, I will try to work on this a little bit in the meantime to speed up things when I return.

**Comment by roryabraham** [comment]
Thanks for the updates, @fabioh8010. For what it's worth, this is not so urgent that I'd want to you to work on it while you're OOO. Enjoy your time off, you've earned it! 🙂

**Comment by roryabraham** [comment]
Sounds like @fabioh8010 will be picking this back up next week

**Comment by fabioh8010** [comment]
Hey! Yes I will go back to this one today.

**Comment by roryabraham** [comment]
Sounds great @fabioh8010, keep me appraised of your progress and let me know if there's anything you need!

**Comment by fabioh8010** [comment]
**Update:** I was able to make some progress of the refactoring and migration of the code to TS, but there are still some functions left to do. Also, I noticed that in the newest version of RN (0.73) the Android files are in Kotlin language now, which the feature is not prepared to deal with. I will need to adjust my feature to detect that and handle it correctly.

**Comment by roryabraham** [comment]
Sounds good @fabioh8010. Without having a ton of context, it seems like it might be helpful to try and split up the work into smaller chunks and potentially try to get more people involved. I leave it up to you of course, but don't hesitate to bring this up in slack if you need support.

**Comment by fabioh8010** [comment]
Sure! I was able to finish TS refactoring last week, so now we only have two things pending:
- Fix the logic to consider Kotlin files when the template is in Kotlin (for newest versions)
- Implement the tests

I will raise a Draft PR today and @TMisiukiewicz will help me this week when available!

**Comment by fabioh8010** [comment]
Draft PR is here -> https://github.com/react-native-community/cli/pull/2276

**Comment by fabioh8010** [comment]
**Update:** Draft PR got some reviews that I will address tomorrow, and will start implementing the tests. @TMisiukiewicz will help me with Kotlin logic.

**Comment by fabioh8010** [comment]
**Update:** Most of the comments addressed, Kotlin files support implemented (thanks @TMisiukiewicz ❤️ ), started implementing unit tests to cover most of the link situations possible.

**Comment by fabioh8010** [comment]
**Update:** Good portion of tests implemented, continuing implementation of the remaining.

**Comment by fabioh8010** [comment]
**Update:** One last test is pending implementation, will add it today together with documentation and open the PR to review.

**Comment by fabioh8010** [comment]
**Update:** Added last test and documentation, PR opened to review. cc @roryabraham

**Comment by fabioh8010** [comment]
**Update:** Still under review process.

**Comment by fabioh8010** [comment]
> @fabioh8010 Do you think it would be possible to back-port the Kotlin support into the CK fork? Or is there any other way you can suggest to use the latest changes while waiting for the PR review?

Sure, I can prepare a new release of the fork on Monday!

**Comment by ramijarrar** [comment]
@fabioh8010 Sounds great - thanks for all your work on this!

**Comment by ramijarrar** [comment]
@fabioh8010 Perfect - thanks again!

**Comment by fabioh8010** [comment]
**Update:** Addressing last comments before merging.

**Comment by fabioh8010** [comment]
**Update:** PR was finally merged 🎉 We should have a new release of the CLI this week!

Given [this discussion](https://github.com/react-native-community/cli/pull/2276#issuecomment-2003394975) and internal conversations, we decided to not expose the link assets as a public command in the cli, but instead keep it as a package.

cc @roryabraham

**Comment by sakluger** [comment]
I'm adding this to the vip-vsb project since all open issues need to be associated with a whatsnext project. Sounds like we're almost done here though, exciting!

**Comment by roryabraham** [comment]
great, looking forward to seeing the E/App PR 🚀

**Comment by fabioh8010** [comment]
Nice, will go back to it this week 🙂

**Comment by luacmartins** [comment]
Getting close, missing the App PR

**Comment by fabioh8010** [comment]
@roryabraham @luacmartins 

A WIP Draft PR is up -> https://github.com/Expensify/App/pull/41115

Still need to do extensive testing on the App, I'm still thinking about an effective way to test those changes besides manual tests as this PR basically affects the whole app and is not feasible to cover all this manually.

Also, just FYI that I'm going to be OOO next week, and will return on May 6th.

**Comment by roryabraham** [comment]
> I'm still thinking about an effective way to test those changes besides manual tests as this PR basically affects the whole app and is not feasible to cover all this manually

One option would be to create an AdHoc build and have Applause do full regression testing against that build.

**Comment by roryabraham** [comment]
What's the latest here?

**Comment by fabioh8010** [comment]
I went back from vacations, working on it today.

**Comment by fabioh8010** [comment]
@roryabraham PR is ready to review

**Comment by zanyrenney** [comment]
sounds good @sakluger

**Comment by fabioh8010** [comment]
**Update:** There is still some feedback I need to address in the PR before removing the WIP label. I plan to address everything next week as this one I need to focus to get the detailed section of a design doc ready to review.

**Comment by fabioh8010** [comment]
**Update:** Applied the last feedbacks and removed the PR from WIP state, review and testing can resume.

**Comment by roryabraham** [comment]
Looks like a number of bugs were reported in the AdHoc testing

**Comment by fabioh8010** [comment]
Reviewed all bugs reported and fixed one bug related to this PR, it's ready to review again.

**Comment by fabioh8010** [comment]
Still under review Melvin

**Comment by roryabraham** [comment]
PR finally merged

**Comment by sakluger** [comment]
I'm pretty sure we don't need any new regression tests here. @mollfpr please request payment via NewDot using the payment summary above. Thanks!

**Comment by mollfpr** [comment]
Thanks @sakluger! Could you confirm if the first amount of bounty applied here?

**Comment by JmillsExpensify** [comment]
$250 approved for @mollfpr
